In [3]:
import os

def count_subfolders_files(folder_path):
    if not os.path.exists(folder_path) or not os.path.isdir(folder_path):
        print("Invalid folder path!")
        return

    for root, dirs, files in os.walk(folder_path):
        folder_name = os.path.relpath(root, folder_path)

        # Counting subfolders and files inside the current folder
        subfolder_count = len(dirs)
        file_count = len(files)

        print(f"Folder Name: {folder_name}")
        print(f"Number of Subfolders: {subfolder_count}")
        print(f"Number of Files: {file_count}")
        print("-" * 30)

# Replace 'path/to/your/folder' with the path of the folder you want to analyze
folder_path = 'train'
count_subfolders_files(folder_path)


Folder Name: .
Number of Subfolders: 100
Number of Files: 0
------------------------------
Folder Name: ADONIS
Number of Subfolders: 0
Number of Files: 126
------------------------------
Folder Name: AFRICAN GIANT SWALLOWTAIL
Number of Subfolders: 0
Number of Files: 107
------------------------------
Folder Name: AMERICAN SNOOT
Number of Subfolders: 0
Number of Files: 105
------------------------------
Folder Name: AN 88
Number of Subfolders: 0
Number of Files: 121
------------------------------
Folder Name: APPOLLO
Number of Subfolders: 0
Number of Files: 128
------------------------------
Folder Name: ARCIGERA FLOWER MOTH
Number of Subfolders: 0
Number of Files: 140
------------------------------
Folder Name: ATALA
Number of Subfolders: 0
Number of Files: 143
------------------------------
Folder Name: ATLAS MOTH
Number of Subfolders: 0
Number of Files: 129
------------------------------
Folder Name: BANDED ORANGE HELICONIAN
Number of Subfolders: 0
Number of Files: 139
--------------

+---------------------------+----------------------+--------------------+
|       Folder Name         | Number of Subfolders | Number of Files    |
+---------------------------+----------------------+--------------------+
| .                         | 100                  | 0                  |
| ADONIS                    | 0                    | 5                  |
| AFRICAN GIANT SWALLOWTAIL  | 0                    | 5                  |
| AMERICAN SNOOT            | 0                    | 5                  |
| AN 88                     | 0                    | 5                  |
| APPOLLO                   | 0                    | 5                  |
| ARCIGERA FLOWER MOTH      | 0                    | 5                  |
| ATALA                     | 0                    | 5                  |
| ATLAS MOTH                | 0                    | 5                  |
| BANDED ORANGE HELICONIAN  | 0                    | 5                  |
| BANDED PEACOCK            | 0                    | 5                  |
| BANDED TIGER MOTH         | 0                    | 5                  |
| BECKERS WHITE             | 0                    | 5                  |
| BIRD CHERRY ERMINE MOTH   | 0                    | 5                  |
| BLACK HAIRSTREAK          | 0                    | 5                  |
| BLUE MORPHO               | 0                    | 5                  |
| BLUE SPOTTED CROW         | 0                    | 5                  |
| BROOKES BIRDWING          | 0                    | 5                  |
| BROWN ARGUS               | 0                    | 5                  |
| BROWN SIPROETA            | 0                    | 5                  |
| CABBAGE WHITE             | 0                    | 5                  |
| CAIRNS BIRDWING           | 0                    | 5                  |
| CHALK HILL BLUE           | 0                    | 5                  |
| CHECQUERED SKIPPER        | 0                    | 5                  |
| CHESTNUT                  | 0                    | 5                  |
| CINNABAR MOTH             | 0                    | 5                  |
| CLEARWING MOTH            | 0                    | 5                  |
| CLEOPATRA                 | 0                    | 5                  |
| CLODIUS PARNASSIAN        | 0                    | 5                  |
| CLOUDED SULPHUR           | 0                    | 5                  |
| COMET MOTH                | 0                    | 5                  |
| COMMON BANDED AWL         | 0                    | 5                  |
| COMMON WOOD-NYMPH         | 0                    | 5                  |
| COPPER TAIL               | 0                    | 5                  |
| CRECENT                   | 0                    | 5                  |
| CRIMSON PATCH             | 0                    | 5                  |
| DANAID EGGFLY             | 0                    | 5                  |
| EASTERN COMA              | 0                    | 5                  |
| EASTERN DAPPLE WHITE      | 0                    | 5                  |
| EASTERN PINE ELFIN        | 0                    | 5                  |
| ELBOWED PIERROT           | 0                    | 5                  |
| EMPEROR GUM MOTH          | 0                    | 5                  |
| GARDEN TIGER MOTH         | 0                    | 5                  |
| GIANT LEOPARD MOTH        | 0                    | 5                  |
| GLITTERING SAPPHIRE       | 0                    | 5                  |
| GOLD BANDED               | 0                    | 5                  |
| GREAT EGGFLY              | 0                    | 5                  |
| GREAT JAY                 | 0                    | 5                  |
| GREEN CELLED CATTLEHEART  | 0                    | 5                  |
| GREEN HAIRSTREAK          | 0                    | 5                  |
| GREY HAIRSTREAK           | 0                    | 5                  |
| HERCULES MOTH             | 0                    | 5                  |
| HUMMING BIRD HAWK MOTH    | 0                    | 5                  |
| INDRA SWALLOW             | 0                    | 5                  |
| IO MOTH                   | 0                    | 5                  |
| Iphiclus sister           | 0                    | 5                  |
| JULIA                     | 0                    | 5                  |
| LARGE MARBLE              | 0                    | 5                  |
| LUNA MOTH                 | 0                    | 5                  |
| MADAGASCAN SUNSET MOTH    | 0                    | 5                  |
| MALACHITE                 | 0                    | 5                  |
| MANGROVE SKIPPER          | 0                    | 5                  |
| MESTRA                    | 0                    | 5                  |
| METALMARK                 | 0                    | 5                  |
| MILBERTS TORTOISESHELL    | 0                    | 5                  |
| MONARCH                   | 0                    | 5                  |
| MOURNING CLOAK            | 0                    | 5                  |
| OLEANDER HAWK MOTH        | 0                    | 5                  |
| ORANGE OAKLEAF            | 0                    | 5                  |
| ORANGE TIP                | 0                    | 5                  |
| ORCHARD SWALLOW           | 0                    | 5                  |
| PAINTED LADY              | 0                    | 5                  |
| PAPER KITE                | 0                    | 5                  |
| PEACOCK                   | 0                    | 5                  |
| PINE WHITE                | 0                    | 5                  |
| PIPEVINE SWALLOW          | 0                    | 5                  |
| POLYPHEMUS MOTH           | 0                    | 5                  |
| POPINJAY                  | 0                    | 5                  |
| PURPLE HAIRSTREAK         | 0                    | 5                  |
| PURPLISH COPPER           | 0                    | 5                  |
| QUESTION MARK             | 0                    | 5                  |
| RED ADMIRAL               | 0                    | 5                  |
| RED CRACKER               | 0                    | 5                  |
| RED POSTMAN               | 0                    | 5                  |
| RED SPOTTED PURPLE        | 0                    | 5                  |
| ROSY MAPLE MOTH           | 0                    | 5                  |
| SCARCE SWALLOW            | 0                    | 5                  |
| SILVER SPOT SKIPPER       | 0                    | 5                  |
| SIXSPOT BURNET MOTH       | 0                    | 5                  |
| SLEEPY ORANGE             | 0                    | 5                  |
| SOOTYWING                 | 0                    | 5                  |
| SOUTHERN DOGFACE          | 0                    | 5                  |
| STRAITED QUEEN            | 0                    | 5                  |
| TROPICAL LEAFWING         | 0                    | 5                  |
| TWO BARRED FLASHER        | 0                    | 5                  |
| ULYSES                    | 0                    | 5                  |
| VICEROY                   | 0                    | 5                  |
| WHITE LINED SPHINX MOTH   | 0                    | 5                  |
| WOOD SATYR                | 0                    | 5                  |
| YELLOW SWALLOW TAIL       | 0                    | 5                  |
| ZEBRA LONG WING           | 0                    | 5                  |
+---------------------------+----------------------+--------------------+


traibn


+-------------------------------+--------------------+
|       Folder Name             | Number of Files    |
+-------------------------------+--------------------+
| .                             | 0                  |
| ADONIS                        | 126                |
| AFRICAN GIANT SWALLOWTAIL      | 107                |
| AMERICAN SNOOT                | 105                |
| AN 88                         | 121                |
| APPOLLO                       | 128                |
| ARCIGERA FLOWER MOTH          | 140                |
| ATALA                         | 143                |
| ATLAS MOTH                    | 129                |
| BANDED ORANGE HELICONIAN      | 139                |
| BANDED PEACOCK                | 118                |
| BANDED TIGER MOTH             | 136                |
| BECKERS WHITE                 | 116                |
| BIRD CHERRY ERMINE MOTH       | 133                |
| BLACK HAIRSTREAK              | 121                |
| BLUE MORPHO                   | 107                |
| BLUE SPOTTED CROW             | 123                |
| BROOKES BIRDWING              | 165                |
| BROWN ARGUS                   | 169                |
| BROWN SIPROETA                | 140                |
| CABBAGE WHITE                 | 127                |
| CAIRNS BIRDWING               | 117                |
| CHALK HILL BLUE               | 149                |
| CHECQUERED SKIPPER            | 136                |
| CHESTNUT                      | 122                |
| CINNABAR MOTH                 | 131                |
| CLEARWING MOTH                | 122                |
| CLEOPATRA                     | 133                |
| CLODIUS PARNASSIAN            | 122                |
| CLOUDED SULPHUR               | 128                |
| COMET MOTH                    | 105                |
| COMMON BANDED AWL             | 123                |
| COMMON WOOD-NYMPH             | 128                |
| COPPER TAIL                   | 133                |
| CRECENT                       | 138                |
| CRIMSON PATCH                 | 103                |
| DANAID EGGFLY                 | 135                |
| EASTERN COMA                  | 132                |
| EASTERN DAPPLE WHITE          | 132                |
| EASTERN PINE ELFIN            | 136                |
| ELBOWED PIERROT               | 117                |
| EMPEROR GUM MOTH              | 136                |
| GARDEN TIGER MOTH             | 131                |
| GIANT LEOPARD MOTH            | 129                |
| GLITTERING SAPPHIRE           | 130                |
| GOLD BANDED                   | 104                |
| GREAT EGGFLY                  | 105                |
| GREAT JAY                     | 135                |
| GREEN CELLED CATTLEHEART      | 126                |
| GREEN HAIRSTREAK              | 176                |
| GREY HAIRSTREAK               | 123                |
| HERCULES MOTH                 | 137                |
| HUMMING BIRD HAWK MOTH        | 143                |
| INDRA SWALLOW                 | 115                |
| IO MOTH                       | 120                |
| Iphiclus sister               | 134                |
| JULIA                         | 115                |
| LARGE MARBLE                  | 116                |
| LUNA MOTH                     | 117                |
| MADAGASCAN SUNSET MOTH        | 123                |
| MALACHITE                     | 103                |
| MANGROVE SKIPPER              | 125                |
| MESTRA                        | 120                |
| METALMARK                     | 108                |
| MILBERTS TORTOISESHELL        | 137                |
| MONARCH                       | 128                |
| MOURNING CLOAK                | 187                |
| OLEANDER HAWK MOTH            | 132                |
| ORANGE OAKLEAF                | 125                |
| ORANGE TIP                    | 137                |
| ORCHARD SWALLOW               | 109                |
| PAINTED LADY                  | 111                |
| PAPER KITE                    | 129                |
| PEACOCK                       | 120                |
| PINE WHITE                    | 123                |
| PIPEVINE SWALLOW              | 119                |
| POLYPHEMUS MOTH               | 123                |
| POPINJAY                      | 121                |
| PURPLE HAIRSTREAK             | 112                |
| PURPLISH COPPER               | 132                |
| QUESTION MARK                 | 110                |
| RED ADMIRAL                   | 117                |
| RED CRACKER                   | 137                |
| RED POSTMAN                   | 127                |
| RED SPOTTED PURPLE            | 123                |
| ROSY MAPLE MOTH               | 137                |
| SCARCE SWALLOW                | 139                |
| SILVER SPOT SKIPPER           | 119                |
| SIXSPOT BURNET MOTH           | 100                |
| SLEEPY ORANGE                 | 152                |
| SOOTYWING                     | 128                |
| SOUTHERN DOGFACE              | 119                |
| STRAITED QUEEN                | 124                |
| TROPICAL LEAFWING             | 118                |
| TWO BARRED FLASHER            | 109                |
| ULYSES                        | 113                |
| VICEROY                       | 115                |
| WHITE LINED SPHINX MOTH       | 143                |
| WOOD SATYR                    | 102                |
| YELLOW SWALLOW TAIL           | 107                |
| ZEBRA LONG WING               | 104                |
+-------------------------------+--------------------+
